### PR TITLE
fix(client): store challenge panes sizes

### LINKED ISSUE #39088: Save width platforms' panels for a client
I think it will be great to have a feature to save the width of instructions-panel, editor-panel, preview-panel for a client in the local storage for example. Every lesson to change it's very boring.

### PR DESCRIPTION
**Pull request by victor-duarte**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #39088

- Base on [Re-Flex Storage demo](https://github.com/leefsmp/Re-Flex/blob/master/src/demo/demo.jsx#L711) example
- Aims to add functionality on desktop view

### Steps to test

1. Open challenge view
2. Resize a pane
3. Either refresh window or go to next challenge

**Ouput**
Window should retain resize made by user

### CODE REVIEW

**Review by raisedadead — CHANGES_REQUESTED** [review]
Thanks for the PR. Mind taking a look at the tests, please?

**Comment by victor-duarte** [comment]
> Thanks for the PR. Mind taking a look at the tests, please?

@raisedadead, a code update has been submitted and test are passing.

Any comments or suggestions are welcome.

**Review by raisedadead — COMMENTED** [review]
Thanks @victor-duarte - I have requested a review from the team and someone will get back to you if further changes are needed. 

Thanks a lot for working on this.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for taking the time to put this together, @victor-duarte. It really is a nice PR.

I have made a few, minor suggestions on things that could be refactored, but I think it's already in a good state.

File: client/src/templates/Challenges/classic/Show.js [hunk 1]
```diff
@@ -92,6 +94,24 @@ const propTypes = {
...
+    flex: 1
+  },
+  editorPane: {
+    flex: 1
+  },
+  instructionPane: {
+    flex: 1
+  },
+  previewPane: {
+    flex: 0.5
```
[the remark this answers is not in the document]
Comment (victor-duarte) [reply client/src/templates/Challenges/classic/Show.js:109]: You are correct. It has been updated to use the original value. 👍🏻

File: client/src/templates/Challenges/classic/Show.js [hunk 2]
```diff
@@ -22,7 +22,9 @@ import DesktopLayout from './DesktopLayout';
 import Hotkeys from '../components/Hotkeys';
 
 import { getGuideUrl } from '../utils';
+import { isBrowser } from '../../../../utils';
```
[the remark this answers is not in the document]
Comment (victor-duarte) [reply client/src/templates/Challenges/classic/Show.js:25]: Gotcha, this is cleaner and consistent.

**Review by ojeytonwilliams — APPROVED** [review]
Thanks for making those changes @victor-duarte 

LGTM :+1:

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
#### :sparkles: :sparkles: **AWESOME! LGTM.** :sparkles: :sparkles: 

Thanks, @victor-duarte, for this pull request and for contributing to the code-base for the first time.

We will be looking forward to more contributions in future.

Cheers and happy contributing!

**Pull request merged**